test/lib/nodes/address.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO(ksuszyns): remove the whole package after knative/pkg#1001 is merged
+
+package nodes
+
+import (
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GuessNodeExternalAddress tries to guess external address of a node
+func (n *NodesClient) GuessNodeExternalAddress(node *corev1.Node) *corev1.NodeAddress {
```

Comment:
I'm not quite understanding why we are twiddling with nodes and internal addresses? We already use Kservice for the forwarder, so why do need these low level details of the node? Is there a reason why we can't just use Kservice for both the forwarder and receiver?

Reply:
This is used to fetch the report. The receiver collects events and prepares a report that lists all errors. Error means: missed event or received twice.

This report is fetch via HTTP to the tests code (outside the cluster).

This code is also proposed in This PR https://github.com/knative/pkg/pull/1001

Reply:
The knative/pkg#1001 was closed. I propose to merge this, and then rewrite getting the report to use k8s etcd infrastructure like writing to configmap and reading from it.